hilti/include/rt/types/map.h

```diff
@@ -87,9 +87,8 @@ class Map : protected std::map<K, V>, public hilti::rt::detail::iterator::Contro
 
         throw IndexError("key does not exist");
     }
-
-    const V& operator[](const K& k) const { return get(k); }
-    auto operator[](const K& k);
+    const V& operator[](const K& k) const { return static_cast<const M&>(*this)[k]; }
```

Comment:
This is leading to unusual semantics now: reading from an non-existing index inserts the element and returns a default constructed values. That's pretty uncommon behavior for scripting languages. However, I don't think there's much we can do about it without having the AssignProxy, and I still favor getting rid of that. But I think we need to document this, should go into the Index operator docs for HILTI's map (and we probably want to expand docs for vector/set there, too; hadn't thought above that earlier).

Reply:
I brought back a slightly rewritten `AssignProxy`. PTAL.